Makefile.include

```diff
@@ -67,34 +67,49 @@ __OVERRIDE_DIRECTORY_VARIABLES := $(__DIRECTORY_VARIABLES)
 # Use absolute paths in recursive "make" even if overridden on command line.
 MAKEOVERRIDES += $(foreach v,$(__OVERRIDE_DIRECTORY_VARIABLES),$(v)=$($(v)))
 
+# Setting EXTERNAL_BOARD_DIRS as command line argument is too messy to handle:
+# Even when every path in EXTERNAL_BOARD_DIRS is turned into an absolute path
+# using override, sub-makes will still get the original value. Using
+# MAKEOVERRIDES has issues with spaces in the values, which are used as
+# separator in EXTERNAL_BOARD_DIRS. So we just enforce setting the value
+# either in a Makefile, or as environment variable.
+ifeq ($(origin EXTERNAL_BOARD_DIRS),command line)
+  $(error EXTERNAL_BOARD_DIRS must be passed as environment variable, and not as command line argument)
+endif
```

Comment:
```suggestion
+ifeq ($(INSIDE_DOCKER),0)
 ifeq ($(origin EXTERNAL_BOARD_DIRS),command line)
   $(error EXTERNAL_BOARD_DIRS must be passed as environment variable, and not as command line argument)
 endif
+endif
```

Reply:
The other option is turning the `error` into a `warning`

Reply:
Done. I changed the order (first check for the origin, than check if running in docker), so that I could add a comment on why this can be allowed when running in docker.

Reply:
I rephrased the comment and squashed that right in.